pageserver/src/consumption_metrics.rs

```diff
@@ -322,3 +355,44 @@ pub async fn collect_metrics_task(
 
     Ok(())
 }
+
+/// Caclculate synthetic size for each active tenant
+pub async fn calculate_synthetic_size_worker(
+    synthetic_size_calculation_interval: Duration,
+) -> anyhow::Result<()> {
+    info!("starting calculate_synthetic_size_worker");
+
+    let mut ticker = tokio::time::interval(synthetic_size_calculation_interval);
+
+    loop {
+        tokio::select! {
+            _ = task_mgr::shutdown_watcher() => {
+                return Ok(());
+            },
+        _ = ticker.tick() => {
+
+                let tenants = mgr::list_tenants().await;
+                // iterate through list of Active tenants and collect metrics
+                for (tenant_id, tenant_state) in tenants {
+
+                    if tenant_state != TenantState::Active {
+                        continue;
+                    }
+
+                    match mgr::get_tenant(tenant_id, true).await
+                    {
+                        Ok(tenant) => {
+                            if let Err(e) = tenant.calculate_synthetic_size().await {
+                                error!("failed to calculate synthetic size for tenant {}: {}", tenant_id, e);
+                            }
+                        },
+                        Err(err) => {
+                            error!("tenant {} is not found: {err:?}", tenant_id);
```

Comment:
I think we could leave this out, since this is inherently racy, the time between the `mgr::list_tenants().await` to this branch might be long.

Also now thinking that this should had been tenant task.